### PR TITLE
Optimizing imports for ktor-utils

### PR DESCRIPTION
**Pull request by mjovanc**
**Motivation**
Removing unused imports and optimizing imports that can be improved using IntelliJ IDEA "Optimize imports".

### CODE REVIEW

**Comment by mjovanc** [comment]
Is the build failure known? I see that commits on `main` branch has failures as well.

**Comment by rsinukov** [comment]
Hi! Do you mind running `./gradlew formatKotlin` to fix the checkstyle and commit it here?

**Comment by rsinukov** [comment]
There are build error on native targets. Can you check it please? You may need to set `jvmAndCommonOnly=false` inside `gradle.properties`

**Comment by mjovanc** [comment]
> There are build error on native and jvm targets. Can you check it please? You may need to set `jvmAndCommonOnly=false` inside `gradle.properties`

Sure. Will check it out right away.

**Comment by mjovanc** [comment]
@rsinukov Rebased and updated. Now it works to run the task assemble without build errors.

**Comment by mjovanc** [comment]
@rsinukov I think it passed now. I changed to use only ktor-utils. I think it's best to split it up for each top level library.

**Comment by rsinukov** [comment]
Hey! Again sorry for the long reply. OOM in stress tests are known issue. Same goes for win and macOS builds.

**Review by rsinukov — APPROVED** [review]

File: ktor-utils/common/src/io/ktor/util/internal/LockFreeLinkedList.kt [hunk 1]
```diff
@@ -11,8 +11,6 @@ package io.ktor.util.internal
  */
 
 import kotlinx.atomicfu.*
-import kotlin.jvm.*
```
Comment (rsinukov) [inline ktor-utils/common/src/io/ktor/util/internal/LockFreeLinkedList.kt:14]: Can you revert it please? The build fails on native without it.

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-utils/common/src/io/ktor/util/internal/LockFreeLinkedList.kt (hunk 1 above)
Comment (mjovanc) [reply ktor-utils/common/src/io/ktor/util/internal/LockFreeLinkedList.kt:14]: Solved.

**Comment by rsinukov** [comment]
Compilation is still failing on win

**Comment by mjovanc** [comment]
> Compilation is still failing on windows

Sorry for the spamming, but I do not have a Windows machine to test it locally. I made some changes now again to see if it passes.

**Comment by mjovanc** [comment]
Added back some uncertain imports to see if the build passes. I have no way of testing it locally on Windows so sorry for the spamming.

**Pull request merged**